msal/src/Microsoft.Identity.Client/ClaimsPrincipalExtension.cs

```diff
@@ -0,0 +1,108 @@
+        /// <returns>A string corresponding to an account identifier as defined in <see cref="Microsoft.Identity.Client.AccountId.Identifier"/></returns>
+        public static string GetMsalAccountId(this ClaimsPrincipal claimsPrincipal)
+        {
+            string userObjectId = claimsPrincipal.FindFirstValue("http://schemas.microsoft.com/identity/claims/objectidentifier");
+            if (string.IsNullOrEmpty(userObjectId))
+            {
+                userObjectId = claimsPrincipal.FindFirstValue("oid");
+            }
+            string tenantId = claimsPrincipal.FindFirstValue("http://schemas.microsoft.com/identity/claims/tenantid");
+            if (string.IsNullOrEmpty(tenantId))
+            {
+                tenantId = claimsPrincipal.FindFirstValue("tid");
+            }
+
+            if (string.IsNullOrWhiteSpace(userObjectId)) // TODO: find a better typed exception
+                throw new ArgumentOutOfRangeException("Missing claim 'http://schemas.microsoft.com/identity/claims/objectidentifier' or 'oid' ");
+
+            if (string.IsNullOrWhiteSpace(tenantId))
+                throw new ArgumentOutOfRangeException("Missing claim 'http://schemas.microsoft.com/identity/claims/tenantid' or 'tid' ");
+
+            string accountId = userObjectId + "." + tenantId;
+            return accountId;
+        }
+
+        /// <summary>
+        /// Get the display name for the signed-in user, based on their claims principal
+        /// </summary>
+        /// <param name="claimsPrincipal">Claims about the user/account</param>
+        /// <returns>A string containing the display name for the user, as brought by Azure AD v1.0 and v2.0 tokens,
+        /// or <c>null</c> if the claims cannot be found</returns>
+        /// <remarks>See https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#payload-claims </remarks>
+        public static string GetDisplayName(this ClaimsPrincipal claimsPrincipal)
+        {
+            // Attempting the claims brought by an Azure AD v2.0 token first
+            string displayName = claimsPrincipal.FindFirstValue("preferred_username");
+
+            // Otherwise falling back to the claims brought by an Azure AD v1.0 token
+            if (string.IsNullOrWhiteSpace(displayName))
+            {
+                displayName = claimsPrincipal.FindFirstValue(ClaimsIdentity.DefaultNameClaimType);
+            }
+            
+            // Finally falling back to name
+            if (string.IsNullOrWhiteSpace(displayName))
+            {
+                displayName = claimsPrincipal.FindFirstValue("name");
+            }
+            return displayName;
+        }
+
+         /// <summary>
+        ///     Return the claim value for the first claim with the specified type if it exists, null otherwise
+        /// </summary>
+        /// <param name="principal"></param>
+        /// <param name="claimType"></param>
```

Comment:
empty params and indentation seems a bit off

Reply:
that was not in my file ? and the argument exception seems wrong?

Reply:
@jmprieur - this was not in your file, indeed. You were using an ASP.NET extension method for FindFirstValue. We definitely don't want to take a dependency on ASP.NET so I wrote my own. I'll delete the comments and fix the arg null.

Reply:
ah :) that explains :) Thanks @bgavrilMS